source/common/router/router.cc

```diff
@@ -222,6 +222,28 @@ Filter::~Filter() {
   ASSERT(!retry_state_);
 }
 
+const FilterUtility::StrictHeaderChecker::HeaderCheckResult
+FilterUtility::StrictHeaderChecker::test(Http::HeaderMap& headers,
+                                         const Http::LowerCaseString& target_header) {
+  HeaderCheckResult r;
+  auto all = Http::Headers::get();
+  if (target_header == all.EnvoyUpstreamRequestTimeoutMs) {
+    return is_integral(headers.EnvoyUpstreamRequestTimeoutMs());
+  } else if (target_header == all.EnvoyUpstreamRequestPerTryTimeoutMs) {
+    return is_integral(headers.EnvoyUpstreamRequestPerTryTimeoutMs());
+  } else if (target_header == all.EnvoyMaxRetries) {
+    return is_integral(headers.EnvoyMaxRetries());
+  } else if (target_header == all.EnvoyRetryOn) {
+    return has_valid_retry_fields(headers.EnvoyRetryOn(),
+                                  &Router::RetryStateImpl::parseRetryOnStrict);
+  } else if (target_header == all.EnvoyRetryGrpcOn) {
+    return has_valid_retry_fields(headers.EnvoyRetryGrpcOn(),
+                                  &Router::RetryStateImpl::parseRetryGrpcOnStrict);
+  }
```

Comment:
I want to double-check my understanding that an assertion here would be safe to do, because Envoy would already fail to start given the Protobuf validations for the router config above?

A softer approach might be to `ENVOY_LOG` here instead of asserting? WDYT?

Reply:
Per our slack discussion: will assert here.